src/base/utils/net.cpp

```diff
@@ -38,5 +40,49 @@ namespace Utils
         {
             return !QHostAddress(ip).isNull();
         }
+
+        bool canParseSubnet(const QString &subnetStr)
+        {
+            Subnet subnet;
+            return canParseSubnet(subnetStr, subnet);
+        }
+
+        bool canParseSubnet(const QString &subnetStr, Subnet subnet)
```

Comment:
weird, why have 2 similar functions?
can someone point me to the discussion if there were any?

Reply:
This one should be `parseSubnet()`.
The function above is for convenience (to avoid declaring redundant variable in place we need just test subnet string representation).

Reply:
https://github.com/qbittorrent/qBittorrent/pull/6779#discussion_r143262019

Reply:
I guess you're missing a ampersand...
`bool parseSubnet(const QString &subnetStr, Subnet &subnet)` makes more sense to me.

Reply:
> This one should be parseSubnet().

I think it's confusing for `parseSubnet` not to return a subnet. Especially so when it somehow returns a boolean. Now `canParseSubnet()` is overloaded, with one variant returning the parsed subnet. This is similar to [QHostAddress::toIPv4Address](https://doc.qt.io/archives/qt-5.5/qhostaddress.html#toIPv4Address).

Reply:
>I think it's confusing for parseSubnet not to return a subnet.

In this case it's better to have:
```c++
Subnet parseSubnet(const QString &subnetStr, bool *ok = nullptr);
```

Reply:
This function doesn't really do any of the parsing; it just checks the result of `QHostAddress::parseSubnet` against the `invalid` subnet. In 3/4 of this function's uses, the return is directly used in a conditional. In the 4th use, the result is passed as the sole parameter to `setEnabled`. Thus I think a function that returns a boolean has more utility.

Reply:
>In this case it's better to have:
Subnet parseSubnet(const QString &subnetStr, bool *ok = nullptr);

+1

Reply:
@Piccirello
```c++
Subnet parseSubnet(const QString &subnetStr, bool *ok = nullptr);
bool canParseSubnet(const QString &subnetStr);
```

Reply:
@Piccirello, don't produce meaningless overloads.
The test functions (with names like `isSomething`, `hasSomething`, `canSomething` and so on) usually don't have any side results.
The conversion functions can have additional result of type bool if its return value can't indicate conversion error.

Reply:
>The conversion functions can have additional result of type bool if its return value can't indicate conversion error.

On a second thought, `Subnet parseSubnet(const QString &subnetStr, bool *ok = nullptr);` can indicate conversion error, then the `bool ok` is redundant?

Reply:
Initially I propose to have this wrapper around QHostAddress::parseSubnet to get rid from incorrect in Utils space `isValidSubnet(Subnet subnet)` function.

Reply:
It seems we can use `QHostAddress::parseSubnet` directly in case we don't need to check its result and have `canParseSubnet` for check-only cases.

Reply:
> On a second thought, Subnet parseSubnet(const QString &subnetStr, bool *ok = nullptr); can indicate conversion error, then the bool ok is redundant?
 
It can only indicate a conversion error via the invalid subnet being returned, which is the entire purpose of this function.

> It seems we can use QHostAddress::parseSubnet directly in case we don't need to check its result and have canParseSubnet for check-only cases.

Any code that doesn't need to check the validity of the result already uses `QHostAddress::parseSubnet` directly.